src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -345,30 +346,29 @@ public GetTransactionModel GetTransaction(string txid)
 
         [ActionName("listaddressgroupings")]
         [ActionDescription("Returns a list of grouped addresses which have had their common ownership made public by common use as inputs or as the resulting change in past transactions.")]
-        public AddressGroupingModel[] ListAddressGroupings()
+        public List<object> ListAddressGroupings()
```

Comment:
why List<object>?
List<AddressGroupModel> sounds appropriate to me, returning raw objects sounds wrong, we wants to have typed return value

p.s.
if `AddressGroupingModel` is only used here, we should change it, and make it inherit from List<AddressGroupModel> and remove its AddressGroups property, everything should be fine

Reply:
Hey @MithrilMan it was done like this to mimic Core's output where the property names needs to be omitted:

```
[
  [
    [
      "address",
      amount
    ]
  ]
]
```

as opposed to (pseudo-ish):

```
[
 addressGroupings:
  [
    addressGrouping:
    [
      address : "address",
      amount : amount
    ]
  ]
]
```

The real issue was the `List<>`.